src/connection.rs

```diff
@@ -47,70 +49,79 @@ impl Connection {
     }
 
     pub fn get(&mut self, key: &str) -> MemcacheResult<Option<(Vec<u8>, u16)>> {
-        try!{ self.stream.write_str(format!("get {}\r\n", key).as_slice()) };
+        try!{ self.stream.write(format!("get {}\r\n", key).as_bytes()) };
         try!{ self.stream.flush() };
-        let result = try!{ self.stream.read_line() };
-        if result.as_slice() == "END\r\n" {
+        let mut line : String = String::new();
+        try! { self.stream.read_line(&mut line) };
+        if line == "END\r\n" {
             return Ok(None);
         }
-        let header: Vec<&str> = result.split(' ').collect();
+        let header: Vec<&str> = line.split(' ').collect();
         if header.len() != 4 || header[0] != "VALUE" || header[1] != key {
             return Err(MemcacheError::ServerError);
         }
         let flags: u16 = match header[2].trim().parse() {
-            Some(flags) => { flags }
-            None => { return Err(MemcacheError::ServerError); }
+            Ok(flags) => { flags }
+            Err(_) => { return Err(MemcacheError::ServerError); }
         };
         let length: usize = match header[3].trim().parse() {
-            Some(length) => { length }
-            None => { return Err(MemcacheError::ServerError); }
+            Ok(length) => { length }
+            Err(_) => { return Err(MemcacheError::ServerError); }
         };
-        let value = try!{ self.stream.read_exact(length) };
-        return Ok(Some((value, flags)));
+        let mut buf : Vec<u8> = Vec::with_capacity(length);
+        //buf.resize(length, 0); Safe option to the code bellow
+        unsafe {
```

Comment:
Solved with unsafe operation. But safe enough provided that the length is known at this time.